### PR TITLE
Better exceptions for unimplemented binary models (BTX etc)

### PR DESCRIPTION
**Pull request by abhisrkckl**
Improved error messages
1. Suggest changing BTX -> BT and DDFWHE -> DDH
2. State that MSS, BT1P etc are not *yet* implemented (see https://tempo.sourceforge.net/ref_man_sections/binary.txt)
3. Default error message otherwise

Also has some sourcery refactors

### CODE REVIEW

File: tests/test_model.py [hunk 1]
```diff
@@ -92,9 +92,6 @@ def test_model():
     # run tempo1 also, if the tempo_utils module is available
     did_tempo1 = False
     with contextlib.suppress(Exception):
-        import tempo_utils
-
-        log.info("Running TEMPO1...")
```
Comment (abhisrkckl) [inline tests/test_model.py:97]: It is not actually running tempo, but reading outputs from a text file.

File: src/pint/models/timing_model.py [hunk 2]
```diff
@@ -3827,6 +3823,26 @@ def search_binary_components(self, system_name):
...
+                raise UnknownBinaryModel(
+                    "`BINARY  DDFWHE` is not supported, but the same model "
+                    "is available as `BINARY  DDH`."
+                )
+            elif system_name in ["MSS", "EH", "H88", "DDT", "BT1P", "BT2P"]:
+                # Binary model list taken from
+                # https://tempo.sourceforge.net/ref_man_sections/binary.txt
+                raise UnknownBinaryModel(
+                    f"`The binary model {system_name} is not yet implemented."
+                )
```
Comment (abhisrkckl) [inline src/pint/models/timing_model.py:3844]: This is the functional change in this PR.

**Comment by dlakaplan** [comment]
This looks reasonable.  Can merge when you are ready.

**Comment by abhisrkckl** [comment]
Yes.. This is ready to go.

**Pull request merged**